src/components/smartSearch/filters/MostActive.tsx

```diff
@@ -6,12 +6,11 @@ import { isFilterWithId } from '../utils';
 import StyledNumberInput from '../inputs/StyledNumberInput';
 import StyledSelect from '../inputs/StyledSelect';
 import TimeFrame from './TimeFrame';
-import { ZetkinSmartSearchFilter } from 'types/zetkin';
-import { ZetkinSmartSearchFilterWithId } from 'types/smartSearch';
+import { FILTER_TYPE, SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
 
 interface MostActiveProps {
-    filter: ZetkinSmartSearchFilterWithId | ZetkinSmartSearchFilter | null;
-    onSubmit: (filter: ZetkinSmartSearchFilterWithId | ZetkinSmartSearchFilter) => void;
+    filter: SmartSearchFilterWithId | ZetkinSmartSearchFilter | { type: FILTER_TYPE };
```

Comment:
Does this actually even ever require a `SmartSearchFilterWithId`? Isn't this component just passing back (in `onSubmit`) the same filter that it received as a prop?

If that's the case, it should be enough to type the prop as the normal `ZetkinSmartSearchFilter` interface, and if there's an extra `id` in there that's fine, but not required by the component.

Reply:
You have a fair point, except it's the other way around: it will only ever be "new" or "with ID", and I think it's not a good idea to use types that we know are wrong because we can ignore the fields.